src/Bounds.cpp

```diff
@@ -547,11 +547,21 @@ class Bounds : public IRVisitor {
 
     void visit(const Load *op) {
         op->index.accept(this);
-        if (interval.is_single_point()) {
-            // If the index is const we can return the load of that index
+        if (interval.is_single_point() && is_one(op->predicate)) {
+            // If the index is const and it is not a predicated load,
+            // we can return the load of that index
+            Expr scalar_pred = const_true();
+            if (const Call *c = op->predicate.as<Call>()) {
```

Comment:
I see, so if it's a cast_mask(vector const true), then it needs to be converted to a cast_mask(scalar const true).

Maybe:

    scalar_pred = op->predicate.type().is_vector() ? extract_lane(op->predicate, 0) : op->predicate;